segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/StorageFactory.java

```diff
@@ -17,4 +20,8 @@
      * Creates a new instance of a Storage adapter.
      */
     Storage createStorageAdapter();
+
+    String getName();
+
+    void initialize(ConfigSetup setup, ScheduledExecutorService executor);
```

Comment:
See my above comment about the FactoryFactory
Who is responsible for shutting down this executor?

Reply:
The creator (i.e. `ServiceBuilder`) class controls the lifetime of the executor. This was the approach before these changes, the PR does not change it.